src/mock/quick_p2p/network.rs

```diff
@@ -218,13 +216,31 @@ impl Queue {
         self.0.push_back(packet)
     }
 
-    fn pop(&mut self) -> Option<Packet> {
-        self.0.pop_front()
-    }
-
     fn is_empty(&self) -> bool {
         self.0.is_empty()
     }
+
+    // This function will pop random msg from the queue.
+    fn pop_random_msg(&mut self, rng: &mut SeededRng) -> Option<Packet> {
+        let first_non_random_msg = self
+            .0
+            .iter()
+            .position(|packet| {
+                if let Packet::Disconnect = packet {
+                    true
+                } else {
+                    false
+                }
+            })
```

Comment:
This is incorrect. It should be something like:

```rust
if let Packet::Message = packet {
    false
} else {
    true
}
```
that is, it should give you the first **non**-message `Packet`. The way it is currently it can still reorder `Message` with other types (say `ConnectSuccess`) which can cause issues (and would not match the real quic-p2p behaviour).

Reply:
Ah, ok.
So, only `Message` can be re-ordered, other Packet shall remain the same.
I will update it.